File: app/src/main/java/com/mytlogos/enterprise/background/room/MultiBaseDao.java
package com.mytlogos.enterprise.background.room;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Update;

import java.util.Collection;

public interface MultiBaseDao<T> extends BaseDao<T> {
    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateBulk(Collection<T> collection);

    @Delete
    void deleteBulk(Collection<T> collection);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertBulk(Collection<T> collection);
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/MultiBaseDao.kt
package com.mytlogos.enterprise.background.room

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Update

interface MultiBaseDao<T> : BaseDao<T> {
    @Update
    suspend fun updateBulk(collection: Collection<T>)

    @Delete
    suspend fun deleteBulk(collection: Collection<T>)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertBulk(collection: Collection<T>)
}